patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -1115,6 +1115,35 @@
  		ClientInitialize();
  		base.Initialize();
  	}
+@@ -5179,6 +_,8 @@
+ 		ItemDropsDB = itemDropDatabase;
+ 		bestiaryDatabase.Merge(ItemDropsDB);
+ 		bestiaryDatabaseNPCsPopulator.AddDropOverrides(bestiaryDatabase);
++		TMLLootDB = new TMLLootDatabase();
++		TMLLootDB.Initialize();
+ 		PylonSystem = new TeleportPylonsSystem();
+ 		ItemDropSolver = new ItemDropResolver(itemDropDatabase);
+ 		ShopHelper = new ShopHelper();
+@@ -5242,7 +_,18 @@
+ 		Chest.SetupTravelShop();
+ 		for (int m = 1; m < 100; m++) {
+ 			shop[m] = new Chest();
+-			shop[m].SetupShop(m);
++
++			if (m >= NPCLoader.shopToNPC.Length)
++				continue;
++
++			int npcType = NPCLoader.shopToNPC[m];
++
++			if (m == 25) { // Painter 2 Shop Special Case
++				shop[m].SetupShop(TMLLootDatabase.GetNPCShopName(npcType, "Decor"), m);
++				continue;
++			}
++
++			shop[m].SetupShop(TMLLootDatabase.GetNPCShopName(npcType), m);
```

Comment:
This lookup code is common in two places, so it should go in `NPCLoader` or the loot database. It should be a function that takes `int shopType` and returns `NPCShop`